test/common/router/config_impl_test.cc

```diff
@@ -2029,5 +2034,229 @@ TEST(RoutePropertyTest, excludeVHRateLimits) {
   EXPECT_TRUE(config_ptr->route(headers, 0)->routeEntry()->includeVirtualHostRateLimits());
 }
 
+TEST(CustomRequestHeadersTest, AddNewHeader) {
```

Comment:
It would be good to have specific tests for the req_header_formatter code that have nothing to do with the route config, since the formatter will get more complicated in the future. so you should have two layers of tests.